BrowserKit/Sources/SiteImageView/FaviconURLProcessing/URLFetcher/HTMLDataRequest.swift

```diff
@@ -30,16 +30,11 @@ struct DefaultHTMLDataRequest: HTMLDataRequest {
 
         let urlSession = URLSession(configuration: configuration)
 
-        return try await withCheckedThrowingContinuation { continuation in
-            urlSession.dataTask(with: url) { data, _, error in
-                guard let data = data,
-                      error == nil
-                else {
-                    continuation.resume(throwing: SiteImageError.invalidHTML)
-                    return
-                }
-                continuation.resume(returning: data)
-            }.resume()
+        do {
+            let (data, _) = try await urlSession.data(from: url)
+            return data
+        } catch {
+            throw SiteImageError.invalidHTML
```

Comment:
Q. When you throw here, how does that look in sentry?
I wonder if it will some in breadcrumbs (doubt it) but will show it as a full on crash 🤔

Reply:
Yeah I doubt it too. There's not a whole lot of logging in the favicon related code. As far as I can tell at the very top of the calling chain you'd just get a nil value from a `try?` call, no events.